### ISSUE TITLE
Adding a visibility level between between x_folder_name and folder_name

### ISSUE THREAD

**Issue by wegry** [atelierdisko/dsk#59]
I've been trying to embed an html doc (index.html) containing an interactive svg in the DSK's default front end.

<img width="1563" alt="screen shot 2019-01-09 at 1 20 39 pm" src="https://user-images.githubusercontent.com/5156873/50899154-9e0a8580-1411-11e9-9aad-723ec0102b63.png">

The doc renders but is unable to pull in a separate CSS file. This file could be inlined, but there are other files pulled in that couldn't be.

Would it be possible to have an api route similar to tree, but instead of json blobs, static files are served? The meta file for an asset folder might hint at the rawness availability.

### An example

```yaml
# assets/meta.yaml
# 🍣 🍙 
# Maybe allows serving of directory (and maybe cascades unless overridden by a child)
served-raw: true    
# ...
```

```css
/* assets/flavor.css */
body {
  background-color: hotpink;
}
```

Hitting `/api/v2/raw/assets/flavor.css` would return the corresponding file.

**Comment by mariuswilms** [comment]
Maybe we could parse SVG files and run them through the same process we use for HTML:
https://github.com/atelierdisko/dsk/blob/master/node_doc_transformer.go#L82

**Comment by wegry** [comment]
I was pleasantly surprised that HTML was supported when I tried it. But there is currently no way to exclude a folder from being visible in the tree that doesn't also make the files inaccessible.

I think that handling file accessibility overrides is a nicer way around [file type restrictions than we currently have.](https://github.com/atelierdisko/dsk/blob/master/node.go#L33) Take that judgment with a grain of salt though, I haven't seen if asset accessibility also affects presence within the tree.

Attempting to normalize paths in input SVG would be the most helpful in conjunction with allowing JS|CSS|'Assets'. Doing this without polluting the host folders for less technical users should be a goal IMO.

The SVG/HTML I'm trying to drop in has the following folder structure

```
├── index.html
├── data.js
└── x_static
    ├── font-600.woff2
    ├── font-400.woff2
    ├── a-little.js
    └── main.css
```

I'd like to avoid it, but if we're forced to go the full inlining route, I think the only thing this app loses is the ability to use local custom fonts. My designers might not be happy. I could be wrong about that though. 

If you folks don't think that this belongs in the main DSK at the moment, maybe there's a way to sidestep the whole issue that I'm not seeing.

**Comment by wegry** [comment]
One day I'll get the name right...Sorry for the spam.

**Comment by mariuswilms** [comment]
We plan to implement this slightly different. We'll make everything inside the DDT accessible, so there are only hidden but no ignored or "secret" assets/nodes. We don't want to complicate things by adding more levels of hidden-ness.

**Comment by wegry** [comment]
How will assets be handled then? Are they still listed by default in the default frontend?

**Comment by ChristophLabacher** [comment]
Assets that are hidden will probably not be listed as "Downloads" in the built-in fronend, but they too will be accessible through the API. Every frontend can decide for itself it/how it wants to list hidden nodes and assets.

**Comment by wegry** [comment]
In addition to the keeping items in the tree regardless of visibility, what about tightening up the restriction for “Downloads” by requiring a certain folder name within as the parent node? Anything under “assets” for example, would be listed in Downloads.

This would replace the existing regex setup currently in place for determining assets.

**Comment by mariuswilms** [comment]
> In addition to the keeping items in the tree regardless of visibility, what about tightening up the restriction for “Downloads” by requiring a certain folder name within as the parent node? Anything under “assets” for example, would be listed in Downloads.

This would break the "folder equals design aspect" core concept of DSK. Sticking to this simple rule makes it easy to explain how DSK works with the DDT.

**Comment by mariuswilms** [comment]
For me the only remaining question is, if we would like to exclude system hidden files, those beginning with a leading `.`?

**Comment by mariuswilms** [comment]
Implemented in 63348ad0ba9d7a24110885016f05da969e860e45. Thanks for bringing this issue up and discussing it with us.
